src/commands/generateThought.ts

```diff
@@ -41,12 +84,73 @@ const generateThought: Command = {
     const simplePath = simplifyPath(state, state.cursor!)
     const thought = getThoughtById(state, head(simplePath))
     if (!thought) return
+
+    // Check if current thought is empty and first child is a URL
+    const isCurrentThoughtEmpty = thought.value === ''
+    const children = getChildrenRanked(state, thought.id)
+    const firstChild = children[0]
+    const isFirstChildURL = firstChild && isURL(firstChild.value)
+
+    if (isCurrentThoughtEmpty && isFirstChildURL) {
+      // Try to fetch webpage title
+      const valuePending = '...'
+
+      // Set to pending while title is being fetched
+      dispatch([
+        updateThoughts({
+          thoughtIndexUpdates: {
+            [thought.id]: {
+              ...thought,
+              value: valuePending,
+              generating: true,
+            },
+          },
+          lexemeIndexUpdates: {},
+          local: false,
+          remote: false,
+          overwritePending: true,
+        }),
+        cursorCleared({ value: true }),
+      ])
+
+      const title = await fetchWebpageTitle(firstChild.value)
+
+      if (title) {
+        // Update thought with the fetched title
+        dispatch([
+          editThought({
+            force: true,
+            oldValue: valuePending,
+            newValue: title,
+            path: simplePath,
+          }),
+          setCursor({ path: state.cursor, offset: title.length }),
+          cursorCleared({ value: false }),
+        ])
+      } else {
+        // If title fetching failed, clear the pending state and leave thought empty
+        dispatch([
+          editThought({
+            force: true,
+            oldValue: valuePending,
+            newValue: '',
+            path: simplePath,
+          }),
+          cursorCleared({ value: false }),
+        ])
+      }
+      return
```

Comment:
Do not define multiple return statements in the same function. You need to have better code reuse.

Reply:
The `return` statement is still there, in the middle of the function. Don't do that.